lib/Parse/ParseDecl.cpp

```diff
@@ -970,7 +971,22 @@ bool Parser::parseDifferentiableAttributeArguments(
   SyntaxParsingContext ContentContext(
       SyntaxContext, SyntaxKind::DifferentiableAttributeArguments);
 
-  // Parse optional differentiation parameters, starting with the 'wrt:' label.
+  // Parse optional differentiation parameters, starting with
+  // the 'linear' label (optinal).
+  if (Tok.is(tok::identifier) && Tok.getText() == "linear") {
```

Comment:
I'd recommend mimicking the logic for parsing `wrt` here:
```cpp
  if (Tok.is(tok::identifier) && Tok.getText() == "linear") {
    linear = true;
    // If no trailing comma or 'where' clause, terminate parsing arguments.
    if (Tok.isNot(tok::comma) && Tok.isNot(tok::kw_where))
      return false;
    if (consumeIfTrailingComma())
      return errorAndSkipToEnd();
  }
```

Does this make sense to you?

Reply:
Offline determined that we needed to change the `consumeIfTrailingComma` lambda and diagnostic string,